### PR TITLE
14517 Submission service implementation

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
- Submission service includes a call to get all local storage items and attempt a post to store data in db.
- #### Http call is currently commented out until implemented
- Tests included for both storage and submission service

### CODE REVIEW

File: pupil-spa/src/app/services/submission/submission.service.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  constructor(private http: Http, private storageService: StorageService) {
+  }
+
+  async submitData() {
+    const localStorageData = this.storageService.getAllItems();
+    const headers = new Headers();
+    headers.append('Content-Type', 'application/json');
+    const requestArgs = new RequestOptions({headers: headers});
+    // TODO: build Check complete API call
```
Comment (GeorgeChatzigiannis) [inline pupil-spa/src/app/services/submission/submission.service.ts]: Will be uncommented once the api call exists and tests will be amended/improved accordingly

File: pupil-spa/src/app/services/question/question.service.ts [hunk 2]
```diff
@@ -55,9 +57,9 @@ export class QuestionService {
 
     if (currentQuestionNumber === this.getNumberOfQuestions()) {
       // we are already on the last question, there isn't another question number.
+      this.submissionService.submitData().catch(error => new Error(error));
```
Comment (jon-shipley) [inline pupil-spa/src/app/services/question/question.service.ts]: Putting the submission service here is incongruous, and violates the single responsibility principle.

Comment (GeorgeChatzigiannis) [reply pupil-spa/src/app/services/question/question.service.ts]: @jon-shipley What's wrong with calling a service from another service?

Comment (GeorgeChatzigiannis) [reply pupil-spa/src/app/services/question/question.service.ts]: I see what you mean but this block really did nothing before apart from exiting.

Comment (jon-shipley) [reply pupil-spa/src/app/services/question/question.service.ts]: @GeorgeChatzigiannis  - so a caller calls `getNextQuestionNumber()` an as a side-effect it submits to the backend?  That's not good.

Comment (jon-shipley) [reply pupil-spa/src/app/services/question/question.service.ts]: @GeorgeChatzigiannis - control flow for the Check itself is managed by the Check Component - the submission must be made from there.

**Pull request merged**